Azure-Samples/communication-services-android-calling-hero
# Remove observers to CallState when leaving a call

## Purpose

* Prevent a race condition, which is triggered when you leave a Teams Interop call
* You may see the "You've been removed from this meeting", or see a stacktrace which indicates that it was unable to create a dialogue for an activity which was already destroyed.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
* Test the code

```
- Ensure that when you leave a Teams Interop meeting that you do not see a dialogue popup showing "You've been removed from this meeting", nor a logging stacktrace which indicates that an Activity was prematurely destroyed.
- Ensure all other CallState events behave as expected(e.g. Be denied joining a meeting, be kicked from the meeting)
```
## Other Information
